man/download-gencode-genome.1.ronn
# download-gencode-genome(1) - download gencode genome

FIXME Need to update the argument names here...

## SYNOPSIS

`download-gencode-genome`
`--organism`=<organism>
[`--build`=<build>]
[`--release`=<release>]
[`--type`=<type>]
[`--annotation`=<annotation>]
[`--output-dir`=<output_dir>]
[`--decompress`]

## DESCRIPTION

Download GENCODE genome and corresponding annotation files.

Creates a new directory in the current working directory.
Only *Homo sapiens* and *Mus musculus* genomes are supported.
The legacy Homo sapiens GRCh37 (hg19) genome build is supported.

## OPTIONS

  * `--organism`=<organism>:
    Latin name (e.g. "Homo sapiens").

  * `--build`=<build>:
    Genome build (e.g. "GRCh38").

  * `--release`=<release>:
    GENCODE release version (e.g. 32).
    If left unset, defaults to current release.

  * `--type`=<type>:
    Genome type:
    `all` (default): Download all reference types;
    `genome`: Primary assembly (DNA);
    `transcriptome`: Transcripts (cDNA/RNA);
    `none`: Skip download.

  * `--annotation`=<annotation>:
    Annotation type:
    `all` (default): Download all annotation formats;
    `gtf`: GTF (GFFv2);
    `gff`: GFF3;
    `none`: Skip download.

  * `--output-dir`=<output_dir>:
    Output directory.
    Defaults to working directory.

  * `--decompress`:
    Decompress files (but keep the compressed originals).

## SEE ALSO

- https://www.gencodegenes.org/
- https://www.gencodegenes.org/pages/faq.html
- https://www.gencodegenes.org/human/release_32lift37.html
- https://github.com/diekhans/gencode-backmap
- tximeta Bioconductor package.

## NOTE

R function, defined in r-koopa package.


NEWS
## koopa 0.3.2 (2019-04-18)

- Improved build scripts, and hardened some intended for RedHat Linux.
- Updated genome build information.
- Improved bcbio scripts and custom genome builds.
- New exported scripts: `ip-address.sh`, `vim-sort.sh`.
- Added script to build Python from source for RedHat.



## koopa 0.3.1 (2019-03-19)

- Updated to Ensembl release to 95.
- Updated GENCODE to release human 29 / mouse M20.
- Updated FlyBase to 2019_01.
- `git-pull-all.sh`: Added `git fetch` call.
- Improved build configuration scripts: emacs, git, gsl, openssl, tmux.
- Added bcbio system configuration files (bcbio_system.yaml).
- Improved default run template for bcbio RNA-seq.



## koopa 0.3.0 (2019-02-20)

This release overhauls the previous activation method. Refer to the README
for details on how to reconfigure your shell profile.

- Changed activation method to use `activate.sh` script at top of repo.
- Improved support for ksh and zsh shells. Added Travis CI checks.
- Added native support for zsh pure prompt, including async script. These files
  are saved in `zsh/fpath/`.
- Reorganized bcbio configuration scripts and templates.
- Completely overhauled system activation method. Additional shell
  configuration, including prompt string customization is now disabled by
  default. Refer to the installation instructions for details on how to enable
  this new optional `extra` mode.
- Reorganized installer scripts exported in `bin/` including conda scripts,
  `reset-permissions.sh`, `sudo-yum-update.sh`, etc.



## koopa 0.2.8 (2019-02-12)

- Migrated darwin (macOS)-specific scripts from bash repo to koopa.
- Migrated Harvard O2 and Harvard Odyssey HPC scripts here from bash repo.
- Added `rsync-azure-files.sh` script for working with Azure Shares (Samba).
- Added `autopad-samples.sh` script.
- Added `bash-strict-mode.sh` script.
- Added `md5sum.sh` script.
- Reorganized other scripts accessible via `bin/`.
- Reorganized activation and system scripts, in preparation for 0.3 release.



## koopa 0.2.7 (2019-02-04)

- Improved shellchecks on workflows.
- Split out some activation steps into separate scripts.



## koopa 0.2.6 (2019-01-23)

- Added back ".sh" extension for all exported scripts.
- Updated Travis CI configuration to use shellcheck.
- Reorganized and improved workflow scripts.



## koopa 0.2.5 (2019-01-07)

- Updated installation instructions to recommend cloning to `~/.koopa`.
- Conda now will only activate when `CONDA_DEFAULT_ENV` is set in environment.
  This improves handling in situations for user accounts where we don't want
  to activate conda in a shared environemnt (e.g. bioinfo account on Azure).



## koopa 0.1.1 (2018-09-05)

- Now exporting `KOOPA_VERSION` and `KOOPA_DATE` global variables.
- Updated installation instructions.
- Removed `/setup/` config files.
- Added support for automatic SSH key import, using `$SSH_KEY` variable.
- Removed "g*" aliases for GitHub commands (e.g. ga, gc, gd). Too confusing.
- Simplified the separator bar for koopa message, avoiding unicode.
- Renamed `bcl2fastq_indrop` script to `bcl2fastq_indrops`.
- Added some utility scripts from macOS bash repo:
      - delete_fs_cruft
      - git_pull_all
      - gzip_dir
      - install_tex
      - move_files_up_1_level
      - rename_from_csv
      - reset_permissions



## koopa 0.1.0 (2018-08-24)

- Initial stable release.
- Repo was previously named "seqcloud".


lang/r/include/download-gencode-genome.R
#!/usr/bin/env Rscript

args <- commandArgs()
whichFile <- grep(pattern = "--file", x = args)
file <- args[whichFile]
file <- sub(pattern = "^--file=", replacement = "", x = file)
koopaPrefix <- normalizePath(file.path(dirname(file), "..", "..", ".."))
source(file.path(koopaPrefix, "lang", "r", "include", "header.R"))

local({
    input <- parseArgs(
        required = c(
            "organism",
            "genome-build"
        ),
        optional = c(
            "release",
            "type",
            "annotation",
            "output-dir"
        ),
        flags = "decompress",
        positional = FALSE
    )
    args <- list()
    args[["organism"]] <- input[["required"]][["organism"]]
    args[["genomeBuild"]] <- input[["required"]][["genome-build"]]
    if (isSubset("release", names(input[["optional"]]))) {
        args[["release"]] <- input[["optional"]][["release"]]
    }
    if (isSubset("type", names(input[["optional"]]))) {
        args[["type"]] <- input[["optional"]][["type"]]
    }
    if (isSubset("annotation", names(input[["optional"]]))) {
        args[["annotation"]] <- input[["optional"]][["annotation"]]
    }
    if (isSubset("output-dir", names(input[["optional"]]))) {
        args[["outputDir"]] <- input[["optional"]][["output-dir"]]
    }
    if (isSubset("decompress", names(input[["flags"]]))) {
        args[["decompress"]] <- TRUE
    }
    requireNamespaces("AcidGenomes")
    do.call(
        what = AcidGenomes::downloadGencodeGenome,
        args = args
    )
})
